teuthology/task/install/__init__.py

```diff
@@ -417,6 +416,21 @@ def ceph_deploy_upgrade(ctx, config):
 ceph_deploy_upgrade.__doc__ = docstring_for_upgrade.format(
     cmd_parameter='ceph_deploy_upgrade')
 
+def _override_extra_system_packages(config, project, install_overrides):
+    teuthology.deep_merge(config, install_overrides.get(project, {}))
+    extra_overrides = install_overrides.get('extra_system_packages')
+    if extra_overrides:
+        extra = config.get('extra_system_packages')
+        if extra is None:
+            e = dict(deb=[], rpm=[])
+        elif isinstance(extra, list):
+            e = dict(deb=copy.deepcopy(extra), rpm=copy.deepcopy(extra))
+        elif isinstance(extra, dict):
+            e = copy.deepcopy(extra)
+        elif isinstance(extra, str):
+            e = dict(deb=[extra], rpm=[extra])
+
+        config['extra_system_packages'] = teuthology.deep_merge(e, extra_overrides)
```

Comment:
This introduced a regression for job definitions where `extra_system_packages` at the top level (i.e. on the install task itself) is a list.  The RBD suite was impacted in a major way.